ui/app/routes/jobs/job/index.js

```diff
@@ -9,12 +10,20 @@ import {
 import WithWatchers from 'nomad-ui/mixins/with-watchers';
 
 export default class IndexRoute extends Route.extend(WithWatchers) {
+  @service can;
+
   async model() {
-    // Optimizing future node look ups by preemptively loading everything
-    await this.store.findAll('node');
     return this.modelFor('jobs.job');
   }
 
+  async afterModel(model) {
+    // Optimizing future node look ups by preemptively loading all nodes if
+    // necessary and allowed.
+    if (model.get('hasClientStatus') && this.can.can('read client')) {
+      await this.store.findAll('node');
```

Comment:
Do users with write access not need this optimization for some reason?

Reply:
I'm not sure if this is how it works in other projects, but in Nomad ACL system a `write` permission assumes that you have `read` as well:
https://github.com/hashicorp/nomad/pull/11672/files#diff-e4c09a2e6416b7748fae6b3d5e3010f68eb770c0b1d757eed0ab8287146bb1b2R18